### PR TITLE
MAINT: update linting setup (align with geopandas, use ruff, move config to pyproject.toml)

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Mostly just copied geopandas' setup (with its latest changes to expand the usage of ruff in https://github.com/geopandas/geopandas/pull/3291 and https://github.com/geopandas/geopandas/pull/3297), with some minor edits to adjust for dask-geopandas

The diff is mostly from sorting imports, and only few actual code changes.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,12 +1,18 @@
+ci:
+    autofix_prs: false
+    autoupdate_schedule: weekly
```
Comment (martinfleis) [inline .pre-commit-config.yaml:3]: Can we do quarterly? Ruff's release frequency otherwise generates update PRs a bit too often to my liking.

Comment (jorisvandenbossche) [reply .pre-commit-config.yaml:3]: Certainly (I just copied it from geopandas)

Comment (jorisvandenbossche) [reply .pre-commit-config.yaml:3]:
```suggestion
    autoupdate_schedule: quarterly
```

Comment (martinfleis) [reply .pre-commit-config.yaml]: Ah. We should update it there as well. In any case, we shall enable geopandas org on pre-commit-ci, right now this does nothing :)

Comment (jorisvandenbossche) [reply .pre-commit-config.yaml]: > right now this does nothing :)

Ah, that's why I couldn't relate with your "generates update PRs a bit too often to my liking" ;)

**Pull request merged**